TeamAmaze/AmazeFileManager
# Fix MainActivity does not respond to view compressed file intents

## Description
- Propagate MIME type when triggering OpenFileDialogFragment
- Detect URI on view intents in MainActivity. If it's a compressed file, triggers MainActivity.openzip()
- Use constant in MimeTypes.ALL_MIME_TYPES for "*/*"

#### Issue tracker   
Fixes #2802, as a band-it fix. Since the open archive dialog assumed supplied path is a local file, hence had to resort to use only CompressedExplorerFragment to handle which supports `content://` URIs (expecting potential problems though).

For a full fix, archive extraction have to support `content://` URIs too; alternately at least open archive dialog should be able to translate `content://` URIs to local files.

#### Manual tests
  
- Device: Pixel 2 emulator
- OS: Android 11
With scenario mentioned in #2802, after selecting Amaze as app to open compressed file Amaze should open the compressed file directly.

#### Build tasks success  

Successfully running following tasks on local:
